src/components/nav-icon-item/nav-icon-item.stories.tsx

```diff
@@ -73,6 +77,22 @@ export const ActiveState: Story = {
   },
 }
 
+export const BadgeState: Story = {
+  args: {
+    icon: <Icon icon="star" />,
+    onClick: action('handleOnClick'),
+    hasBadge: true,
+  },
+  render: (args) => {
+    return (
+      <ElButtonNavIconItem onClick={action('handleClick')}>
+        {args?.icon}
+        <ElNavIconItemBadge role="status" />
+      </ElButtonNavIconItem>
+    )
+  },
+}
+
```

Comment:
**thought:** I find it a bit strange that we have so many "style-only usage" examples here. Personally, I think our stories should focus more on the React usage, but I guess we can sweep through the docs in future and tidy them up. Right now, I just don't think we're going far enough with our docs; they seem to be a bit of an afterthought.